src/python/pants/backend/visibility/rule_types.py

```diff
@@ -264,27 +264,28 @@ def _get_address_path(address: Address) -> str:
 
     def get_action(
         self,
-        target: TargetAdaptor,
         address: Address,
-        relpath: str,
+        adaptor: TargetAdaptor,
+        other_address: Address,
     ) -> tuple[VisibilityRuleSet | None, DependencyRuleAction | None, str | None]:
         """Get applicable rule for target type from `path`.
 
         The rules are declared in `relpath`.
         """
-        ruleset = self.get_ruleset(address, target, relpath)
+        relpath = self._get_address_relpath(address)
+        ruleset = self.get_ruleset(address, adaptor, relpath)
```

Comment:
It is the `address` passed to `get_ruleset()` here that has been swapped. The other changes are due to argument renames to make them more comprehensible, and moved the callsite for getting the `relpath` value.